### PR TITLE
[Travis] Add Ruby 2.3 to matrix

### PR DESCRIPTION
**Pull request by segiddins**
Blocked by https://github.com/travis-ci/travis-ci/issues/5361.

### CODE REVIEW

**Comment by segiddins** [comment]
OK, only issue here is different versions of Psych using different quotes. Need to figure out a good way to transform this without totally ruining the value of the tests

**Comment by segiddins** [comment]
@mrackwitz let me know if you're OK with these changes?

File: spec/integration.rb [hunk 1]
```diff
@@ -91,6 +91,18 @@
...
 
+  c.preprocess('**/*.xcodeproj', %r{(^|/)(Podfile|Manifest).lock$}) do |path|
+    keys_hint = if path.extname == '.lock'
+                  Pod::Lockfile::HASH_KEY_ORDER
+                end
+    contents = path.read
+    if contents.strip.empty?
+      contents
+    else
+      Pod::YAMLHelper.convert_hash(YAML.load(path.read), keys_hint, "\n\n")
```
Comment (mrackwitz) [inline spec/integration.rb]: `s/path.read/contents/`

**Comment by segiddins** [comment]
:shipit:

**Pull request merged**